### PR TITLE
Add native_max_local_exchange_partition_count session property

### PR DESCRIPTION
**Pull request by arhimondr**
## Description

Add native_max_local_exchange_partition_count session property

## Motivation and Context

Maps to the max_local_exchange_partition_count Velox query property
introduced in https://github.com/facebookincubator/velox/pull/11292

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [X] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [X] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [X] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [X] Adequate tests were added if applicable.
- [X] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== RELEASE NOTES ==

Prestissimo (Native Execution) Changes
* Add ``native_max_local_exchange_partition_count session`` property. :pr:`23910`
```

### CODE REVIEW

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeHiveExternalTableTpchQueries.java
Comment (xiaoxmeng) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeHiveExternalTableTpchQueries.java:117]: Remove before land? Thanks!

**Review by xiaoxmeng — APPROVED** [review]
@arhimondr accepted with nit. Thanks!

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeHiveExternalTableTpchQueries.java
Comment (arhimondr) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeHiveExternalTableTpchQueries.java:117]: I commented the following line to unblock. Created an issue to resolve it properly: https://github.com/prestodb/presto/issues/23908

Let me leave a comment on the issue to remove this comment once resolved.

Comment (arhimondr) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeHiveExternalTableTpchQueries.java:117]: https://github.com/prestodb/presto/issues/23908#issuecomment-2444804185

**Review by xiaoxmeng — APPROVED** [review]

**Comment by steveburnett** [comment]
Suggest adding documentation for this session property to [Presto C++ Session Properties](https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/presto_cpp/properties-session.rst). 

Also, minor nit to include the PR number in the release note entry

```
== RELEASE NOTES ==

Native
* Add ``native_max_local_exchange_partition_count session`` property :pr:`23910`
```

File: presto-docs/src/main/sphinx/presto_cpp/properties-session.rst
Comment (steveburnett) [inline presto-docs/src/main/sphinx/presto_cpp/properties-session.rst:270]:
```suggestion
```
Suggest deleting this sentence because it repeats the information in line 266 and does not express the number in a different way that is more easily human-readable. For example, line 260 restates the default value for ``native_max_page_partitioning_buffer_size`` from ``33554432`` to ``32MB``.

**Review by steveburnett — CHANGES_REQUESTED** [review]
Thanks for the doc! Minor suggestion for the content. Please let me know, and maybe suggest a rephrasing, if you feel this sentence or one like it helps the reader.

File: presto-docs/src/main/sphinx/presto_cpp/properties-session.rst
Comment (arhimondr) [reply presto-docs/src/main/sphinx/presto_cpp/properties-session.rst]: Good catch. Fixed.

**Review by steveburnett — APPROVED** [review]
LGTM! (docs)

Pull updated branch, new local doc build, looks good. Thanks!

**Review by xiaoxmeng — APPROVED** [review]

**Review by NikhilCollooru — APPROVED** [review]

**Pull request merged**
